[BUG] Timestamp test cases failed due to Python time zone is not UTC

**Describe the bug**
Some timestamp cases generate `9999-12-31 18:19:13` (UTC) timestamp, which causes `datetime.fromtimestamp` out-of-range error if the local time zone is not UTC like `Beijing +8 hour` time zone.

When generating test data set, the error occurs: 
```
ValueError: year 0 is out of range
```

The following commands do not help, because the error is thrown in Python.
```
export PYSP_TEST_spark_driver_extraJavaOptions=-Duser.timezone=UTC
export PYSP_TEST_spark_executor_extraJavaOptions=-Duser.timezone=UTC
export PYSP_TEST_spark_sql_session_timeZone=UTC
```

So we should set the timezone on the Python side when creating test data set.

**Steps/Code to reproduce bug**
Set local time zone to `+8:00`  timezone

ENV: 
- Spark 331 is from: https://urm.nvidia.com/artifactory/sw-spark-maven/org/apache/spark/3.3.1/
- Rapids branch is: branch-23.02
- Python 3.8.12 (default, Oct 12 2021, 13:49:34)

Compile: 
env -u SPARK_HOME mvn clean install -Drat.skip=true -Dmaven.javadoc.skip=true -Dskip -Dmaven.scalastyle.skip=true -DskipTests -pl dist -am -DallowConventionalDistJar=true -Dbuildver=331

Run test case:
./integration_tests/run_pyspark_from_build.sh -s -k test_explode_array_data

```
_________ test_explode_array_data[Map(Timestamp(not_null),Timestamp)] __________

data_gen = [Integer, Array(Map(Timestamp(not_null),Timestamp))]

    @ignore_order(local=True)
    @pytest.mark.order(1) # at the head of xdist worker queue if pytest-order is installed
    @pytest.mark.parametrize('data_gen', explode_gens + struct_gens_sample_with_decimal128 +
                             array_gens_sample + map_gens_sample + arrays_with_binary + maps_with_binary,
                             ids=idfn)
    def test_explode_array_data(data_gen):
        data_gen = [int_gen, ArrayGen(data_gen)]
>       assert_gpu_and_cpu_are_equal_collect(
            lambda spark: two_col_df(spark, *data_gen).selectExpr('a', 'explode(b)'),
            conf=conf_to_enforce_split_input)

../../src/main/python/generate_expr_test.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/main/python/asserts.py:548: in assert_gpu_and_cpu_are_equal_collect
    _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
../../src/main/python/asserts.py:467: in _assert_gpu_and_cpu_are_equal
    run_on_cpu()
../../src/main/python/asserts.py:453: in run_on_cpu
    from_cpu = with_cpu_session(bring_back, conf=conf)
../../src/main/python/spark_session.py:113: in with_cpu_session
    return with_spark_session(func, conf=copy)
../../src/main/python/spark_session.py:97: in with_spark_session
    ret = func(_spark)
../../src/main/python/asserts.py:201: in <lambda>
    bring_back = lambda spark: limit_func(spark).collect()
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/dataframe.py:818: in collect
    return list(_load_from_socket(sock_info, BatchedSerializer(CPickleSerializer())))
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/serializers.py:151: in load_stream
    yield self._read_with_length(stream)
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/serializers.py:173: in _read_with_length
    return self.loads(obj)
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/serializers.py:471: in loads
    return cloudpickle.loads(obj, encoding=encoding)
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:1729: in <lambda>
    return lambda *a: dataType.fromInternal(a)
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:821: in fromInternal
    values = [
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:822: in <listcomp>
    f.fromInternal(v) if c else v
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:594: in fromInternal
    return self.dataType.fromInternal(obj)
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:518: in fromInternal
    return obj and dict(
/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:519: in <genexpr>
    (self.keyType.fromInternal(k), self.valueType.fromInternal(v)) for k, v in obj.items()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = TimestampType(), ts = 253402280353328528

    def fromInternal(self, ts: int) -> datetime.datetime:
        if ts is not None:
            # using int to avoid precision loss in float
>           return datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
E           ValueError: year 0 is out of range

/home/remziy/working/spark/spark-3.3.1-bin-hadoop3/python/lib/pyspark.zip/pyspark/sql/types.py:223: ValueError
```

**Expected behavior**

Set Python time zone to UTC before run test cases.

**Additional context**
$SPARK_HOME/bin/pyspark
```
SparkSession available as 'spark'.
>>> import time
>>> import datetime
>>> os.environ['TZ'] = 'Asia/Hong_Kong'
>>> time.tzset()
>>> ts = 253402280353328528
>>> datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: year 0 is out of range
>>> os.environ['TZ'] = 'UTC'
>>> time.tzset()
>>> datetime.datetime.fromtimestamp(ts // 1000000).replace(microsecond=ts % 1000000)
datetime.datetime(9999, 12, 31, 18, 19, 13, 328528)
>>>
```

Note: If set local timezone to UTC, this test case can pass.

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -25,6 +25,13 @@ from spark_session import is_tz_utc, is_before_spark_340
 import sre_yield
 import struct
 from conftest import skip_unless_precommit_tests
+import time
+import os
+
+# set time zone to UTC for timestamp test cases to avoid `datetime` out-of-range error:
+# refer to: https://github.com/NVIDIA/spark-rapids/issues/7535
+os.environ['TZ'] = 'UTC'
+time.tzset()
 
 class DataGen:
     """Base class for data generation"""

